pkg/controller/persistentvolume/persistentvolume_index.go

```diff
@@ -117,16 +92,20 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 				continue
 			}
 
-			if claim.Name == volume.Spec.ClaimRef.Name && claim.Namespace == volume.Spec.ClaimRef.Namespace && claim.UID == volume.Spec.ClaimRef.UID {
-				// exact match! No search required.
+			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
Don't see how to add this comment to persistentvolume_controller.go where it belongs since that file in this pr doesn't contain the func isVolumeBoundToClaim. Sorry. Anyway, looking at:
```
func isVolumeBoundToClaim(volume *api.PersistentVolume, claim *api.PersistentVolumeClaim) bool {...
        if volume.Spec.ClaimRef == nil {
                return false
        }
        if claim.Name != volume.Spec.ClaimRef.Name || claim.Namespace != volume.Spec.ClaimRef.Namespace {
                return false
        }
        if volume.Spec.ClaimRef.UID != "" && claim.UID != volume.Spec.ClaimRef.UID {
                return false
        }
        return true
}
```
I think this func should also handle the case of a pvc pre-binding to a pv via `**spec.VolumeName**`, eg:
```
        ...
        if claim.Spec.VolumeName != "" && claim.spec.VolumeName != volume.Name {
                return false
        }
        return true
}
```